productRouter.js

```diff
@@ -0,0 +1,67 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const {Product} = require('./product.model');
+const {extend} = require("lodash");
+
+router.route("/")
+.get(async (req, res) => {
+  const products = await Product.find({});
+  res.json({products, success: true})
+})
+.post( async (req, res) => {
+  try {
+  const { name, numberOfProducts } = req.body;
+
+  const NewProduct = new Product({name: name, numberOfProducts: numberOfProducts});
+
+  const savedProduct = await NewProduct.save();
+
+  console.log(savedProduct);
+
+  res.json({savedProduct, success: true});
+
+  } catch(err) {
+    res.status(500).json({ success:false, message: "failed to add product", errorMessage: err.message});
+  }
+  
```

Comment:
Just a suggestion here, you can try keeping all these db related tasks in a separate folder like controllers , so the code will look more cleaner.